tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/SpecValidationTool/SpecValidationTool.cs

```diff
@@ -0,0 +1,128 @@
+            {
+                validationResults.Add($"TypeSpec project is not found in {typeSpecProjectRootPath}. TypeSpec MCP tools can only be used for TypeSpec based spec projects.");
+                return validationResults;
+            }
+
+            try
+            {
+                var specRepoRootPath = GetGitRepoRootPath(typeSpecProjectRootPath);
+                _logger.LogInformation($"Repo root path: {specRepoRootPath}");
+                // Run npm ci
+                _logger.LogInformation("Running npm ci");                
+                RunNpmCi(specRepoRootPath);
+                _logger.LogInformation("Completed runnign npm ci");
+
+                //Run TypeSpec validation
+                _logger.LogInformation("Running npx tsv to run the validation");
+                ValidateTypeSpec(typeSpecProjectRootPath, specRepoRootPath, validationResults);
+                _logger.LogInformation("Completed running TypeSpec validation");
+            }
+            catch (Exception ex)
+            {
+                validationResults.Add($"Error: {ex.Message}");
+            }
+            return validationResults;
+        }
+
+        public static void RunNpmCi(string repoRoot)
+        {
+            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+            if (isWindows)
+            {
+                _ = RunProcess("cmd.exe", "/C npm ci", repoRoot);
+            }
+            else
+            {
+                _ = RunProcess("npm", "ci", repoRoot);
+            }
+        }
+
+        private static string ValidateTypeSpec(string typeSpecProjectRootPath, string specRepoRootPath, IList<string> validationResults)
+        {
+            var isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+            if (isWindows)
+            {                
+                var output = RunProcess("cmd.exe", $"/C npx tsv {typeSpecProjectRootPath}", specRepoRootPath);
+                validationResults.Add(output);
+            }
+            else
+            {
+                var output = RunProcess("npx", $"tsv {typeSpecProjectRootPath}", specRepoRootPath);
+                validationResults.Add(output);
+            }
+            return "TypeSpec validation completed successfully";
+        }
+       
+
+        private static string GetGitRepoRootPath(string typeSpecProjectRootPath)
+        {
+            var currentDirectory = new DirectoryInfo(typeSpecProjectRootPath);
+            while (currentDirectory != null && !currentDirectory.Name.Equals("specification", StringComparison.OrdinalIgnoreCase))
+            {
+                currentDirectory = currentDirectory.Parent;
+            }
+            return currentDirectory?.Parent?.FullName ?? string.Empty;
+        }
+
+        private static string RunProcess(string command, string args, string workingDirectory)
+        {
+            var processInfo = new ProcessStartInfo
```

Comment:
I will pull across my process helper from `test-proxy` at some point. Will refactor this for that when I do.

Reply:
Yup. We need a common process running helper class. I had started on that but didn't include it later.